LSP: `linked_editing_range` uses current window for refresh

### Problem

`tsgo` crashes when enabling linked editing range via `vim.lsp.linked_editing_range.enable(true)` and the server is enabled / started from a buffer whose filetype doesn't match tsgo's config.

Note: **also affects Typescript 7's official LSP.**

Related: https://github.com/neovim/neovim/issues/36646 (kind of?)

### Steps to reproduce using "nvim --clean -u minimal_init.lua"

- repro.lua

```lua
vim.lsp.linked_editing_range.enable(true)

vim.lsp.config('tsgo', {
    cmd = { "/path/to/tsgo", "--lsp", "--stdio" },
    filetypes = { 'typescript', },
    root_markers = { 'pnpm-lock.yaml' },
})

vim.lsp.enable('tsgo')
```

- steps
1. `nvim --clean -u repro.lua -o pnpm-lock.yaml some-ts-file.ts` (**note:** the order is important here, the non typescript file must come first)

### Expected behavior

server does not crash

### Nvim version (nvim -v)

v0.13.0-dev-1006+g31ad7d4527

### Language server name/version

tsgo 7.0.0-dev.20260630.1

### Operating system/version

Linux 7.1.4

### Log file

_No response_

Stack from tsc:

```
panic: ScriptKind must be specified when parsing source file: /path/to/non-ts-file [recovered, repanicked]
goroutine 2643 [running]:
sync.(*WaitGroup).Go.func1.1()
	sync/waitgroup.go:251 +0x45
panic({0xd43460?, 0x252a57fe6ed0?})
	runtime/panic.go:860 +0x13a
github.com/microsoft/typescript-go/internal/parser.(*Parser).initializeState(0x252a58db0de0?, {{0x252a57255607, 0x2d}, {0x252a57255607, 0x2d}, {0x1, 0x0}}, {0x252a5a9ca000?, 0x252a58ecc3c0?}, 0x0)
	github.com/microsoft/typescript-go/internal/parser/parser.go:292 +0x459
github.com/microsoft/typescript-go/internal/parser.ParseSourceFile({{0x252a57255607, 0x2d}, {0x252a57255607, 0x2d}, {0x1, 0x0}}, {0x252a5a9ca000, 0x2509}, 0x0)
	github.com/microsoft/typescript-go/internal/parser/parser.go:140 +0x10a
github.com/microsoft/typescript-go/internal/project.NewSession.NewParseCache.func3({{{0x252a57255607, 0x2d}, {0x252a57255607, 0x2d}, {0x1, 0x0}}, 0x0, {0xa74e6695c9fa417f, 0xe989ca52a02d0741}}, {0x117bb60, ...})
	github.com/microsoft/typescript-go/internal/project/parsecache.go:34 +0xb0
github.com/microsoft/typescript-go/internal/project.(*RefCountCache[...]).Acquire(0x6186a5, {{{0x252a57255607, 0x2d}, {0x252a57255607, 0x2d}, {0x1, 0x0}}, 0x0, {0xa74e6695c9fa417f, 0xe989ca52a02d0741}}, ...)
	github.com/microsoft/typescript-go/internal/project/refcountcache.go:49 +0x123
github.com/microsoft/typescript-go/internal/project.(*compilerHost).GetSourceFile(0x252a59bb8aa0, {{0x252a57255607, 0x2d}, {0x252a57255607, 0x2d}, {0x1, 0x0}})
	github.com/microsoft/typescript-go/internal/project/compilerhost.go:100 +0x1c9
github.com/microsoft/typescript-go/internal/compiler.(*fileLoader).parseSourceFile(0x252a58cc5208, 0x252a59de1b00)
	github.com/microsoft/typescript-go/internal/compiler/fileloader.go:370 +0x25b
github.com/microsoft/typescript-go/internal/compiler.(*parseTask).load(0x252a59de1b00, 0x252a58cc5208)
	github.com/microsoft/typescript-go/internal/compiler/filesparser.go:112 +0x4a6
github.com/microsoft/typescript-go/internal/compiler.(*filesParser).start.func1()
	github.com/microsoft/typescript-go/internal/compiler/filesparser.go:290 +0x351
github.com/microsoft/typescript-go/internal/core.(*parallelWorkGroup).Queue.func1()
	github.com/microsoft/typescript-go/internal/core/workgroup.go:40 +0x13
sync.(*WaitGroup).Go.func1()
	sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 18
	sync/waitgroup.go:238 +0x73
```

--- a/runtime/lua/vim/lsp/linked_editing_range.lua
+++ b/runtime/lua/vim/lsp/linked_editing_range.lua
@@ -195,7 +195,9 @@ function LinkedEditor:on_attach(client_id)
     self.client_state[client_id] = state
   end
 
-  self:refresh()
+  if self.bufnr == api.nvim_get_current_buf() then
+    self:refresh()
+  end
 end
 
 ---@package

